subprojects/shipkit/src/main/groovy/org/shipkit/internal/gradle/util/Optional.java

```diff
@@ -0,0 +1,30 @@
+package org.shipkit.internal.gradle.util;
+
+import java.util.NoSuchElementException;
+
+/**
+ * We need to migrate to Java8 and remove this.
+ */
+public class Optional<T> {
+
+    private final T value;
+
+    public Optional(T value) {
+        this.value = value;
+    }
+
+    public T get() {
+        if (value == null) {
+            throw new NoSuchElementException("No value present");
+        }
+        return value;
+    }
+
+    public static <T> Optional<T> of(T object) {
```

Comment:
It looks like this `Optional.of` behaves different than `java.util.Optional.of`. This one is fine with `null` as parameter and the java.util one will throw a NPE.
We have two options:
 - adjust the behavior now (which will allow us to do a more smooth java8 migration)
 - adjust it later on when we migrate to java8

Reply:
I'll fix it now. Good catch!